src/tests/test_certstor_system.cpp

```diff
@@ -49,6 +49,37 @@ Test::Result find_certificate_by_pubkey_sha1(Botan::Certificate_Store& certstore
    return result;
    }
 
+Test::Result find_certificate_by_pubkey_sha1_with_unmatching_key_id(Botan::Certificate_Store& certstore)
+   {
+   Test::Result result("System Certificate Store - Find Certificate by SHA1(pubkey) - regression test for GH #2779");
+
+   if(!certstore.find_cert(get_dn_of_cert_with_different_key_id(), {}).has_value())
+      {
+      result.note_missing("OS does not trust the certificate used for this regression test, skipping");
+      return result;
+      }
```

Comment:
This was required, because AppVeyor doesn't come with any of the "vulnerable" certificates. Hence, on AppVeyor the regression test won't run. The MinGW stage provides `SecureTrust CA`, though.